scanner/src/main/kotlin/Scanner.kt

```diff
@@ -714,20 +715,23 @@ class Scanner(
 
         val duration = measureTime {
             provenancesWithMissingArchives.forEach { (pkg, nestedProvenance) ->
-                runCatching {
-                    downloadRecursively(nestedProvenance)
-                }.onSuccess { dir ->
+                var dir: File? = null
+
+                // runCatching has a bug with smart-cast, see https://youtrack.jetbrains.com/issue/KT-62938.
+                try {
+                    dir = downloadRecursively(nestedProvenance)
                     archiver.archive(dir, nestedProvenance.root)
-                    dir.safeDeleteRecursively(force = true)
-                }.onFailure {
+                } catch (e: IOException) {
                     controller.addIssue(
                         pkg.id,
                         Issue(
                             source = "Downloader",
                             message = "Could not create file archive for " +
-                                "'${pkg.id.toCoordinates()}': ${it.collectMessages()}"
+                                "'${pkg.id.toCoordinates()}': ${e.collectMessages()}"
```

Comment:
Do you think "Downloader" is the right "source" for the issue, when the exception comes from the archiver?
/ Would scanner be a better fit?

Would it make sense to create dedicated issues for both cases, like the following?
1. Could not download nested provenance for ${pkg.id.toCoordinates()
2. Could not create file archive for ${pkg.id.toCoordinates()